### PR TITLE
Update SQL Query for agent_installs that have no labels

### PR DESCRIPTION
**Pull request by harageth**
# Resolves

https://jira.rax.io/browse/SALUS-737

# What

Make sure that we match the agent_installs to that have no labels to everything

# How

Make sure that we use a LEFT OUTER JOIN and then check for NULL values 

## How to test

Run the Unit tests

# Why

Because it needs to happen at query time

# TODO

Should be done

### CODE REVIEW

File: src/test/java/com/rackspace/salus/acm/services/AgentInstallServiceTest.java [hunk 1]
```diff
@@ -145,6 +145,8 @@ public void testGetInstallsFromLabels() {
     // different labelSelectorMethod - only one label
     final AgentInstall install8 = saveInstall(
         release1, "t-1", LabelSelectorMethod.OR, "os", "linux");
+    final AgentInstall install9 = saveInstall(
```
Comment (itzg) [inline src/test/java/com/rackspace/salus/acm/services/AgentInstallServiceTest.java]: An inline comment would be good here that emphasizes that an empty label selector is being declared on "install9" so it will match all of the resource variants below.

File: src/test/java/com/rackspace/salus/acm/services/AgentInstallServiceTest.java [hunk 2]
```diff
@@ -241,7 +243,7 @@ public void testGetInstallsFromLabels() {
           .map(AgentInstall::getId)
           .collect(Collectors.toList());
       assertThat(installIds).containsExactlyInAnyOrder(
-          install8.getId()
+          install8.getId(), install9.getId()
```
Comment (jjbuchan) [inline src/test/java/com/rackspace/salus/acm/services/AgentInstallServiceTest.java]: Do we need one more test with an empty `resourceLabels` map?

Comment (harageth) [reply src/test/java/com/rackspace/salus/acm/services/AgentInstallServiceTest.java]: Currently the code ends up throwing an illegal argument exception. This would be a good time to change that to match whats happening in MonitorManagement and then add these tests.

Comment (harageth) [reply src/test/java/com/rackspace/salus/acm/services/AgentInstallServiceTest.java]: updated
